kodi_addon_checker/check_dependencies.py

```diff
@@ -132,6 +132,32 @@ def check_reverse_dependencies(report: Report, addon: str, branch_name: str, all
                           .format(", ".join(sorted([r.id for r in rdependsLowerBranch])), len(rdependsLowerBranch))))
 
 
+def check_circular_dependencies(report: Report, all_repo_addons: dict, parsed_xml, branch_name: str):
+    """Check for any circular dependencies in addon.xml file and reports them
+        :all_repo_addons: dictionary return by all_repo_addon() function
+        :parsed_xml: parsed addon.xml file
+        :branch_name: name of the kodi branch/version
+    """
+    addon = Addon(parsed_xml)
+    ignore = _get_ignore_list(branch_name)
+
+    dependencies = []
+    for dependency in addon.dependencies:
+        if dependency.id in ignore or dependency.id in VERSION_ATTRB:
```

Comment:
`VERSION_ATTRB` doesn't tell anything about repo availability. Optional dependencies have to be excluded.
```suggestion
        if dependency.id in ignore or dependency.optional:
```

Reply:
We should check `VERSION_ATTRB` entries for circular?
I would think you'd still want to check if an optional dependency is circular because that would be an issue if the user has it installed?

Reply:
`VERSION_ATTRB` is just to give hints on updating the imported version of `xbmc.python` (and in future maybe others). It doesn't define if an addon exists or not. That's defined by `_get_ignore_list(...)` and the add-ons available in the repo.

If you want to check optional dependencies too, I'm fine with that. Those wouldn't have the ordering during add-on installation (if it is correctly implemented in Kodi).

Reply:
I completely missed `xbmc.python` in the ignore list, which is why I used the in `VERSION_ATTRB`.
Will remove that.

Reply:
Removed `in VERSION_ATTRB` now